File: core/src/main/scala/tripaint/grid/ImageGridColorLookup.scala
package tripaint.grid

import tripaint.ColorLookup
import tripaint.color.Color
import tripaint.coords.{GlobalPixCoords, PixelCoords}

class ImageGridColorLookup(grid: ImageGrid) extends ColorLookup {
  override def lookup(coords: GlobalPixCoords): Option[Color] = {
    val pixelCoords = PixelCoords(coords, grid.imageSize)
    val img = grid(pixelCoords.image)
    if img != null then {
      Some(img.storage.getColor(pixelCoords.pix))
    } else {
      None
    }
  }
}

File: core/src/main/kotlin/tripaint/grid/ImageGridColorLookup.kt
package tripaint.grid

import tripaint.ColorLookup
import tripaint.color.Color
import tripaint.coords.GlobalPixCoords
import tripaint.coords.PixelCoords

class ImageGridColorLookup(val grid: ImageGrid) : ColorLookup {
    override fun lookup(coords: GlobalPixCoords): Color? {
        val pixelCoords = PixelCoords.from(coords, grid.imageSize)
        val img = grid.apply(pixelCoords.image)

        return img?.storage?.getColor(pixelCoords.pix)
    }
}